client/lib/signup/step-actions/index.js

```diff
@@ -1206,6 +1212,14 @@ export function isPlanFulfilled( stepName, defaultDependencies, nextProps ) {
 		);
 		recordExcludeStepEvent( stepName, defaultDependencies.cartItem );
 		fulfilledDependencies.push( 'cartItem' );
+	} else if ( isTransferSelectedInDomainTransferFlow ) {
+		const cartItem = null;
+		submitSignupStep(
+			{ stepName, cartItem, wasSkipped: true },
+			{ cartItem, ...dependenciesFromDefaults }
+		);
+		recordExcludeStepEvent( stepName, sitePlanSlug );
+		fulfilledDependencies.push( 'cartItem' );
```

Comment:
Looks more robust indeed. A lot of things to be put together. Makes me wonder how different/simpler such scenario would be in Stepper :-)